packages/compass-import-export/src/modules/export.ts

```diff
@@ -281,12 +284,18 @@ export const selectFieldsToExport = (): ExportThunkAction<
     });
 
     const {
-      export: { query, namespace },
+      export: { query, namespace, connectionId },
     } = getState();
 
     let gatherFieldsResult: Awaited<ReturnType<typeof gatherFieldsFromQuery>>;
 
     try {
+      const dataService =
+        connectionsManager.getDataServiceForConnection(connectionId);
+      if (!dataService) {
+        throw new Error(`DataService for connection ${connectionId} not found`);
+      }
+
```

Comment:
perhaps we can also add a check for connectionId, since it starts as an empty string?